### ISSUE TITLE
Add support for swift's .swiftinterface 

### ISSUE THREAD

**Issue by jerrymarino** [bazel-ios/rules_ios#210]
Add the ability to handle .swiftinterface in rules_ios. Currently this feature requires library evolution but has already been added in rules_swift.

**Comment by brentleyjones** [comment]
Does Swift 5.5's incremental imports solve this in a way that doesn't require using library evolution?

**Comment by keith** [comment]
Definitely, and it doesn't seem like swiftinterface and library evolution are going to be decoupled any time soon, so that should probably be the path forward

**Comment by jerrymarino** [comment]
The swift 5.5 stuff looks great but I'm not 100% sure about how that's going to interop with Bazel and swift. Today it looks like because the `.swiftmodule` is an input, it will cause the actions to re-run. Does this fix that?

Today for remote caching/exec and incremental builds `.swiftinterface` is attractive to prevent the downloads / invalidations overall.

**Comment by keith** [comment]
Yes it fixes that, swiftmodules won't change up the tree if the incremental logic decides it doesn't need to recompile anything

**Comment by jerrymarino** [comment]
That's pretty cool!

The other thing I was hoping for in this feature was improving swift concurrency: making it more like objc and header files. With swift there is a serial chain of swift modules: `swift_library(name=a, deps[b]), swift_library(b)` where `a` has to wait for `b`.

Fixing concurrency may be a pipe-dream but having the interface decoupled from code-gen seems like a step in the right direction.

**Comment by dflems** [comment]
This [doc](https://swift.org/blog/library-evolution/) says library evolution makes some trade-offs with performance. Does anyone know what this means in a more concrete sense? I guess it does fewer optimizations, inlining, etc?
